static/app/components/forms/fields/sentryMemberTeamSelectorField.tsx

```diff
@@ -77,11 +85,36 @@ function SentryMemberTeamSelectorField({
     loadMore: loadMoreTeams,
   } = useTeams({provideUserTeams: true});
 
-  const teamOptions = teams?.map(team => ({
-    value: `team:${team.id}`,
-    label: `#${team.slug}`,
-    leadingItems: <Avatar team={team} size={avatarSize} />,
-  }));
+  // TODO(davidenwang): Fix the team type here to avoid this type cast
+  const teamOptions = (teams as DetailedTeam[])
+    .map(team => {
+      const isDisabledTeam =
+        memberOfProjectSlug &&
+        !defined(team.projects.find(({slug}) => memberOfProjectSlug === slug));
```

Comment:
Do we need `defined` can we just use `some` here?
```suggestion
        !team.projects.some(({slug}) => memberOfProjectSlug === slug);
```

Reply:
ahh yes this is better